radsat-sk/operation/subsystems/RPdb.c

```diff
@@ -0,0 +1,272 @@
+static uint32_t PdbVoltageCommandBytes[4] = {
+		0x10E280,		// Output Voltage of BCR
+		0x10E220,		// Output Voltage of Battery Bus
+		0x10E210,		// Output Voltage of 5V Bus
+		0x10E200		// Output Voltage of 3.3V Bus
+};
+
+/**
+ * 4 Byte commands for requesting output current readings for each bus and BCR
+ */
+static uint32_t PdbCurrentCommandBytes[4] = {
+		0x10E284,		// Output Current of BCR in mA
+		0x10E224,		// Output Current of Battery Bus
+		0x10E214,		// Output Current of 5V Bus
+		0x10E204		// Output Current of 3.3V Bus
+};
+
+
+/***************************************************************************************************
+                                       PRIVATE FUNCTION STUBS
+***************************************************************************************************/
+
+static uint16_t PdbI2cTalk(uint32_t command);
+
+
+/***************************************************************************************************
+                                             PUBLIC API
+***************************************************************************************************/
+/**
+ * Get the Irradiance value from the sun sensor on each face of the CubeSat
+ *
+ * @return A pointer to a SunSensorStatus object containing the converted ADC data for each sun sensor
+ */
+SunSensorStatus getSunSensorData(void) {
+
+	// Create a new SunSensorStatus Structure
+	SunSensorStatus sunData;
+
+	// Create a temporary variable for receiving I2C data
+	uint16_t i2c_received;
+
+	// Create a temporary array for calculated sun data before transferring into the structure sunData
+	float convertedData[NUM_SUN_SENSORS];
+
+	// Send 6 commands to get each of the sun sensor's data
+	for(int i = 0; i < NUM_SUN_SENSORS; i = i + 1){
+		i2c_received = PdbI2cTalk(PdbSunSensorCommandBytes[i]);
+		convertedData[i] = ADC_COUNT_TO_IRRADIANCE * i2c_received;
+	}
+
+	// Now store all of the calculated data into the proper slot in the sunData structure
+	sunData.xPos = convertedData[0];
+	sunData.xNeg = convertedData[1];
+	sunData.yPos = convertedData[2];
+	sunData.yNeg = convertedData[3];
+	sunData.zPos = convertedData[4];
+	sunData.zNeg = convertedData[5];
+	// Could change these numbers into an ENUM with the names of the connections
+	/** TODO: Add Timestamp data into sunData  */
+
+	return sunData;
+}
+
+
+
+/**
+ * Request and store all of the relevant data (voltage, current, temperature, sun sensor data)
+ *  	from the PDB board
+ *
+ * @return A pointer to an PDBStatus object containing all of the telemetry for the PDB system
+ */
+PdbStatus getPdbTelemetry(void) {
+
+	// Create a new PDBStatus Structure
+	PdbStatus dataStorage;
+
+	// Get sun sensor data and store it in the new PDBStatus Structure
+	dataStorage.sunSensorData = getSunSensorData();
+
+	// Create a temporary variable for receiving I2C data
+	uint16_t i2c_received;
+
+
+	// Get ADC output voltage readings from the PDB
+	i2c_received = PdbI2cTalk(PdbVoltageCommandBytes[0]);
```

Comment:
You might want to add an error check or else you will be loading your struct with error messages